### ISSUE TITLE
Legacy mode for intervals

### ISSUE THREAD

**Issue by stdkoehler** [d8ahazard/sd_dreambooth_extension#665]
**WHAT**
Provide an option to activate the legacy mode of selecting number of training steps + checkpoint saving based on training steps + sample image generation per training step

**WHERE**
Input - Settings Tab

**HOW**
Same as in older versions (maybe add a checkbox or dropdown for such a "legacy mode" to avoid the clutter of additional options)

**WHY**
I got a lot more control with that option. When I've got a lot of training pictures for a style (100s), I need to make Training Steps per Image very low to get any early preview at all (since now I'm limited to saving checkpoints and preview based on hole epochs)

Previously, I typically trained 1000 steps with a higher learning rate, then repeated with a lower learning rate and was very flexible in terms of saving intermediate checkpoints and preview images.

This flexibility is now gone and my training results are worse.

**Comment by d8ahazard** [comment]
Ugh...I'll have to think about this one. After spending days trying to work out the proper way to count "steps per epoch", I came to the realization that "steps" is really a pretty arbitrary number, which means - IMHO - it doesn't mean anything. As an epoch is always one iteration over the dataset (or should be), it's a much more rational measure of training progress.

A few thoughts for your situation:

You can just adjust "save image every" to 1 epoch, which should still give you pretty fast repetition of sample images.

There's also the manual "generate samples" button, which can be used any time, but again, will wait till the end of the current epoch, as triggering it otherwise can cause an OOM.

As far as learning rate/repetition - there are multiple new schedulers for LR you should look at. My personal favorite is "cosine annealing" which will start at one learning rate, and then start shifting to a lower lr at a specified point in training. Effectively, it'll do exactly what you're doing now without stop/start again.

**Comment by Claxiz** [comment]
I too, like the ability to set my own learn rate as an option. What's not going so great for me though is the saving options, i have to manually save checkpoints and samples, despite what you suggest, setting the save every X epoch at 1 does nothing for either option on my end. I can't be at my computer all the time so the automated saving of checkpoints is a massive lifesaver for me.

**Comment by stdkoehler** [comment]
> Ugh...I'll have to think about this one. After spending days trying to work out the proper way to count "steps per epoch", I came to the realization that "steps" is really a pretty arbitrary number, which means - IMHO - it doesn't mean anything. As an epoch is always one iteration over the dataset (or should be), it's a much more rational measure of training progress.

Thanks for the heads up. Makes sense regarding the epoch being the smallest unit then. However, I still have been able to produce better results using my previous method. Even though it might not be scientifically bulletproof since it would likely stop in the middle of an epoch before completely finishing the last iteration over the dataset.

I will try the tips you provided with different kind or LR schedulers. If it was possible to still have such a legacy mode I would really appreciate it though.

**Comment by alenknight** [comment]
> 

i'm having the same issue - where no matter what i set in the save to / save every epoch (i was testing with 3 for preview and ckpt), along with enabling the feature in the "save" tab.... it works the first run through for a ckpt... but if you cancel, or even if complete and try again with a new model... it won't stick.

**Comment by d8ahazard** [comment]
Save frequencies are fixed with eb51824. I have no plans on re-implementing anything regarding steps at this point in time.
